ndcube/utils/cube.py

```diff
@@ -119,3 +123,88 @@ def convert_extra_coords_dict_to_input_format(extra_coords, missing_axis):
                 raise KeyError("extra coords dict can have keys 'wcs axis' or 'axis'.  Not both.")
             result.append((name, axis, extra_coords[name]["value"]))
         return result
+
+
+def _get_pixel_quantities_for_dependent_axes(dependent_axes, cube_dimensions):
+    """
+    Returns list of othogonal pixel quantities for cube axes which have dependent WCS translations.
+
+    Parameters
+    ----------
+    dependent_axis: `list` of `int`
+        Axis numbers in the numpy convention.
+
+    cube_dimensions: Iterable of `int`
+        Number of pixels in each dimension
+
+    Returns
+    -------
+    quantity_list: `list` of `astropy.units.Quantity`
+        Othogonal pixel quantities describing all pixels in cube in relevant dimensions.
+
+    """
+    n_dependent_axes = len(dependent_axes)
+    n_dimensions = len(cube_dimensions)
+    quantity_list = [u.Quantity(np.zeros(tuple(cube_dimensions[dependent_axes])),
+                                unit=u.pix)] * n_dimensions
+    for i, dependent_axis in enumerate(dependent_axes):
+        # Define list of indices/slice objects for accessing
+        # sections of dependent axis arrays that are to be
+        # replaced with orthogonal np.arange arrays.
+        slice_list = [0] * n_dependent_axes
+        slice_list[i] = slice(None)
+        # Define array of indices of axes in slice list not
+        # including the axis along which we are inserting
+        # np.arange.
+        other_axes_indices = list(range(n_dependent_axes))
+        del(other_axes_indices[i])
```

Comment:
`other_axes_indices.pop(i)`?

Reply:
Does it make a difference in this case?  ```pop``` is good for keeping the deleted element in another variable.  But in this case we just want it gone entirely.